packages/web3-eth/src/eth_tx.ts

```diff
@@ -28,38 +22,33 @@ import {
 	PopulatedUnsignedTransaction,
 	Transaction,
 } from './types';
-import { getBlock, getGasPrice, getTransactionCount } from './rpc_method_wrappers';
 import { validateChainInfo, validateCustomChainInfo, validateGas } from './validation';
-import { formatTransaction } from './format_transaction';
 
-export const detectTransactionType = (
-	transaction: Transaction,
-	overrideMethod?: (transaction: Transaction) => HexString | undefined,
-): HexString | undefined => {
-	// TODO - Refactor overrideMethod
-	if (overrideMethod !== undefined) return overrideMethod(transaction);
+export const defaultTransactionTypeParser: TransactionTypeParser = transaction => {
+	const tx = transaction as unknown as Transaction;
 
-	if (transaction.type !== undefined)
-		return convertToValidType(transaction.type, ValidTypes.HexString) as HexString;
+	if (tx.type !== undefined)
+		return convertToValidType(tx.type, ValidTypes.HexString) as HexString;
 
 	if (
-		transaction.maxFeePerGas !== undefined ||
-		transaction.maxPriorityFeePerGas !== undefined ||
-		transaction.hardfork === 'london' ||
-		transaction.common?.hardfork === 'london'
+		tx.maxFeePerGas !== undefined ||
+		tx.maxPriorityFeePerGas !== undefined ||
+		tx.hardfork === 'london' ||
+		tx.common?.hardfork === 'london'
 	)
 		return '0x2';
 
-	if (
-		transaction.accessList !== undefined ||
-		transaction.hardfork === 'berlin' ||
-		transaction.common?.hardfork === 'berlin'
-	)
+	if (tx.accessList !== undefined || tx.hardfork === 'berlin' || tx.common?.hardfork === 'berlin')
 		return '0x1';
 
 	return undefined;
 };
 
+export const detectTransactionType = (
+	transaction: Transaction,
+	transactionTypeParser: TransactionTypeParser = defaultTransactionTypeParser,
+) => transactionTypeParser(transaction as unknown as Record<string, unknown>);
```

Comment:
```suggestion
) =>
    // TODO: When we have `web3-types` package we can share TransactionType
    transactionTypeParser(transaction as unknown as Record<string, unknown>);
```

Just so this doesn't get overlooked